common/include/pcl/conversions.h

```diff
@@ -180,11 +180,14 @@ namespace pcl
     cloud.points.resize (num_points);
     uint8_t* cloud_data = reinterpret_cast<uint8_t*>(&cloud.points[0]);
 
-    // Check if we can copy adjacent points in a single memcpy
+    // Check if we can copy adjacent points in a single memcpy.  We can do so if there
+    // is exactly one field to copy and it is the same size as the source and destination
+    // point types.
     if (field_map.size() == 1 &&
         field_map[0].serialized_offset == 0 &&
         field_map[0].struct_offset == 0 &&
-        msg.point_step == sizeof(PointT))
+        field_map[0].size == msg.point_step &&
+        field_map[0].size == sizeof(PointT))
```

Comment:
I think the comment makes clear that the source msg, destination type and field must all be the same size.  Happy to get a consensus from others though.

BTW the Travis checks seem to have failed because compiling the unit tests exceeded the allowed time.  Is that normal?